tests/client/test_async_client.py

```diff
@@ -301,3 +301,22 @@ async def test_mounted_transport():
         response = await client.get("custom://www.example.com")
         assert response.status_code == 200
         assert response.json() == {"app": "mounted"}
+
+
+@pytest.mark.usefixtures("async_environment")
+async def test_response_aclose_map_exceptions():
+    class BrokenStream:
+        async def __aiter__(self):
+            # so we're an AsyncIterator
+            pass  # pragma: nocover
+
+        async def aclose(self):
+            raise httpcore.CloseError(OSError(104, "Connection reset by peer"))
+
+    def handle(request: httpx.Request) -> httpx.Response:
+        return httpx.Response(200, stream=BrokenStream())
+
+    async with httpx.AsyncClient(transport=httpx.MockTransport(handle)) as client:
+        async with client.stream("GET", "http://example.com") as response:
+            with pytest.raises(httpx.CloseError):
+                await response.aclose()
```

Comment:
Am I correct saying this would also reproduce for a regular `client.request()`? If so, should we update this to…?

```suggestion
        with pytest.raises(httpx.CloseError):
            await client.request("GET", "http://example.com")
    
        with pytest.raises(httpx.CloseError):
            async with client.stream("GET", "http://example.com"):
                pass  # Close on exit.
```

(Also dropped the explicit `response.aclose()` since the expected behavior is that exiting the `stream()` async context manager would do that for us, and fail with the proper exception.)

Reply:
You are correct. And the `await client.request()` test is a worthy addition.

You're the maintainer, so you are welcome to change to nix the `aclose()` in the test ... but first, I'll explain my motivation.

During debugging, I was surprised by #1464 -- a broken stream raises the "wrong" error on shutdown.

I imagine a reasonable `__aexit__()` might _handle_ an errno=104 error -- by doing nothing and raising nothing. So I aimed for an explicit test that avoids `__aexit__()`.

The change would be _correct_. But as a user, I find `__axit__()` behavior confusing; so I wrote a more-explicit test.

Reply:
Yup, I like the way @adamhooper has approached the test case here, too.